gpu4pyscf/nac/tdrks.py

```diff
@@ -0,0 +1,346 @@
+    cphf_conv_tol = getattr(__config__, "grad_tdrhf_Gradients_cphf_conv_tol", 1e-8)
+
+    to_cpu = utils.to_cpu
+    to_gpu = utils.to_gpu
+    device = utils.device
+
+    _keys = {
+        "cphf_max_cycle",
+        "cphf_conv_tol",
+        "mol",
+        "base",
+        "chkfile",
+        "states",
+        "atmlst",
+        "de",
+        "de_scaled",
+        "de_etf",
+        "de_etf_scaled",
+        "_write"
+    }
+
+    def __init__(self, td):
+        self.verbose = td.verbose
+        self.stdout = td.stdout
+        self.mol = td.mol
+        self.base = td
+        self.states = (0, 1)  # of which the gradients to be computed.
+        self.atmlst = None
+        self.de = None
+        self.de_scaled = None
+        self.de_etf = None
+        self.de_etf_scaled = None
+
+    _write      = rhf_grad_cpu.GradientsBase._write
+
+    def dump_flags(self, verbose=None):
+        log = logger.new_logger(self, verbose)
+        log.info("\n")
+        log.info(
+            "******** LR %s gradients for %s ********",
+            self.base.__class__,
+            self.base._scf.__class__,
+        )
+        log.info("cphf_conv_tol = %g", self.cphf_conv_tol)
+        log.info("cphf_max_cycle = %d", self.cphf_max_cycle)
+        log.info("chkfile = %s", self.chkfile)
+        log.info(f"States ID = {self.states}")
+        log.info("\n")
+        return self
+
+    @lib.with_doc(get_nacv.__doc__)
+    def get_nacv(self, x_yI, EI, singlet, atmlst=None, verbose=logger.INFO):
+        return get_nacv(self, x_yI, EI, singlet, atmlst, verbose)
+
+    def kernel(self, xy_I=None, xy_J=None, E_I=None, E_J=None, singlet=None, atmlst=None):
+
+        logger.warn(self, "This module is under development!!")
+
+        if singlet is None:
+            singlet = self.base.singlet
+        if atmlst is None:
+            atmlst = self.atmlst
+        else:
+            self.atmlst = atmlst
+
+        if self.verbose >= logger.WARN:
+            self.check_sanity()
+        if self.verbose >= logger.INFO:
+            self.dump_flags()
+
+        if xy_I is None or xy_J is None:
+            states = sorted(self.states)
+            I, J = states
+            if I < 0 or J < 0:
+                raise ValueError("Excited states ID should be non-negetive integers.")
+            elif I > 0:
+                raise NotImplementedError("Only for ground-excited states nonadiabatic coupling.")
+            elif I == 0:
+                xy_I = self.base.xy[J-1]
+                E_I = self.base.e[J-1]
+                self.de, self.de_scaled, self.de_etf, self.de_etf_scaled \
+                    = self.get_nacv(xy_I, E_I, singlet, atmlst, verbose=self.verbose)
+                self._finalize()
+            else:
+                raise NotImplementedError("Only for ground-excited states nonadiabatic coupling.")
+        return self.de
+    
+    def get_veff(self, mol=None, dm=None, j_factor=1.0, k_factor=1.0, omega=0.0, hermi=0, verbose=None):
```

Comment:
Why effective potential contribution in RKS is the same to the one in RHF? Shouldn't the XC part be computed within this function?